src/DocumentDB.ChangeFeedProcessor/FeedProcessing/IPartitionProcessorFactory.cs

```diff
@@ -15,8 +16,9 @@ public interface IPartitionProcessorFactory
         /// Creates an instance of a <see cref="IPartitionProcessor"/>.
         /// </summary>
         /// <param name="lease">Lease to be used for partition processing</param>
+        /// <param name="leaseCheckpointer">Default lease checkpointer</param>
         /// <param name="observer">Observer to be used</param>
         /// <returns>An instance of a <see cref="IPartitionProcessor"/>.</returns>
-        IPartitionProcessor Create(ILease lease, IChangeFeedObserver observer);
+        IPartitionProcessor Create(ILease lease, ILeaseCheckpointer leaseCheckpointer, IChangeFeedObserver observer);
```

Comment:
breaking change

Reply:
Does it worth a new interface? It is just additional parameter passed, looks like a minor breaking change that is easy to adapt. But of course, that can be added if you think it is required.

Reply:
I think let’s not break it. People who implemented this interface would have their own checkpointing, so ignoring new param would be confusing

Reply:
updated with the new interface. I am not sure about the naming though, do you have better suggestions?